### ISSUE TITLE
systemd-networkd: ActivationPolicy ignored in VLANs

### ISSUE THREAD

**Issue by TCB13** [systemd/systemd#22593]
Hello,

I've found that systemd-networkd ignores `ActivationPolicy` in VLANs. This behavior has been going on since the feature was introduced here https://github.com/systemd/systemd/issues/3031. Besides now it was also reported here:

https://superuser.com/questions/1678714/configure-systemd-network-vlans-that-dont-auto-start

> For some reason, neither ActivationPolicy=manual nor ActivationPolicy=down shut the interfaces down; instead, systemd-networkd just ignores it on VLANs and leaves the interface up.

Here is an example configuration I used to test this:

**10-eth0.network:**
````
[Match]
Name=eth0

[Link]
RequiredForOnline=yes

[Network]
DHCP=v4
VLAN=vlan18
````
**vlan18.netdev:**
````
[NetDev]
Name=vlan18
Kind=vlan

[VLAN]
Id=18
````
**vlan18.network:**
````
[Match]
Name=vlan18

[Network]
Address=10.10.1.1/24

[Link]
ActivationPolicy=manual
````

**Used distribution**
Debian GNU/Linux 11 (bullseye)

**Linux kernel version used** (`uname -a`)
`5.10.0-11-amd64 #1 SMP Debian 5.10.92-1 (2022-01-18) x86_64 GNU/Linux`

**CPU architecture issue was seen on**
AMD 64

**Expected behaviour you didn't see**
Set `ActivationPolicy=manual` on the VLAN definition and after boot the interface is down / no IP address present:
````
4: vlan18@eth0: <BROADCAST,MULTICAST> mtu 1500 qdisc noqueue state DOWN group default qlen 1000
    link/ether XX:XX:XX:XX:XX:XXbrd ff:ff:ff:ff:ff:ff
````

**Unexpected behaviour you saw**
The interface is up with the configured IP:
`````
4: vlan18@eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether XX:XX:XX:XX:XX:XXbrd ff:ff:ff:ff:ff:ff
    inet 10.200.1.1/24 brd 10.200.1.255 scope global vlan18
       valid_lft forever preferred_lft forever
`````

**Steps to reproduce the problem**
1. Create VLAN
2. Set `ActivationPolicy=manual` or `ActivationPolicy=down`
3. Reboot the system

Thank you.

**Comment by yuwata** [comment]
Fix is waiting in #22611.

**Comment by TCB13** [comment]
> Fix is waiting in #22611.

Thank you for the fix.
